x-pack/plugins/maps/public/components/validated_number_input.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, ChangeEvent, MouseEvent } from 'react';
+import { EuiFieldNumber, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+
+interface State {
+  value: number | string;
+  errorMessage: string;
+  isValid: boolean;
+}
+
+interface Props {
+  initialValue: number;
+  min: number;
+  max: number;
+  onChange: (value: number) => void;
+  label: string;
+}
+
+function isNumberValid(value: number | string, min: number, max: number) {
+  const parsedValue = parseInt(value.toString(), 10);
+
+  if (isNaN(parsedValue)) {
+    return {
+      isValid: false,
+      errorMessage: i18n.translate('xpack.maps.validatedNumberInput.invalidNumberErrorMessage', {
+        defaultMessage: 'Invalid number',
+      }),
+    };
+  }
+
+  const isValid = parsedValue >= min && parsedValue <= max;
+  return {
+    parsedValue,
+    isValid,
+    errorMessage: isValid
+      ? ''
+      : i18n.translate('xpack.maps.validatedNumberInput.invalidClampErrorMessage', {
+          defaultMessage: 'Value must be between {min} and {max}',
```

Comment:
How about "Must be between {min} and {max}" to be the same as other inputs
<img width="300" alt="Screen Shot 2020-12-14 at 11 13 31 AM" src="https://user-images.githubusercontent.com/373691/102119115-dab13980-3dfd-11eb-8c5e-c4f67220ea63.png">